playwright-junit/src/main/java/com/microsoft/playwright/PlaywrightCleanupExtension.java

```diff
@@ -0,0 +1,31 @@
+package com.microsoft.playwright;
+
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+import static com.microsoft.playwright.BrowserExtension.closeAllBrowsers;
+import static com.microsoft.playwright.ExtensionUtils.getExecutionMode;
+import static com.microsoft.playwright.PlaywrightExtension.closePlaywright;
+
+public class PlaywrightCleanupExtension implements AfterEachCallback, AfterAllCallback {
```

Comment:
We can but then the registration of extensions comes into play regarding which AfterAllCallback hook runs first (we would want browser then playwright in that order). Which is okay since we are registering extensions with our custom annotation.